pandera/schema_components.py

```diff
@@ -63,6 +63,10 @@ def coerce_dtype(self, series):
             (including time series).
 
         """
+        # If no dtype is specified, then assume no coercion is wanted
```

Comment:
Is this actually what we want or would it be better to throw an exception in this case?

Reply:
I think we should maintain the design philosophy of failing early and explicitly. In this case, I would suggest raising a `SchemaInitError` in the following cases:

- if `pandas_dtype` is None and `coerce == True` when defining a `Column`
- if `DataFrameSchema(..., coerce=True)` and at least one of the specified columns has `pandas_dtype is None`.

Let's also add to the `:raises SchemaInitError:` docstring, where this exception is raised when `pandas_dtype is None` and `coerce == True` (and similar docstring in `DataFrameSchema`)

Reply:
Yeah - that makes more sense to me too. Is something like this what you had in mind?